File: 프로그래머스/1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

long long solution(long long n) {
    vector<int> n_arr;
    long long answer = 0;
    
    while (n > 0) {
        n_arr.push_back(n%10);
        n/=10;
    }
    sort(n_arr.rbegin(), n_arr.rend());
    
    string new_number = "";
    vector<int>::iterator ptr;
    for(ptr = n_arr.begin(); ptr != n_arr.end(); ptr ++) {
        new_number += to_string(*ptr);
    }
    
    return stoll(new_number);
}

File: 프로그래머스/1/12933. 정수 내림차순으로 배치하기/정수 내림차순으로 배치하기.js
function solution(n) {
    var answer = 0;
    let number_arr = [];
 
    while (n > 0) {
        number_arr.push(n%10);
        n/=10;
        n=~~n;
    }
  
    number_arr = number_arr.sort((a,b)=>b-a);
    
    for (let i=0; i<number_arr.length; i++){
        answer += number_arr[i];
        if (i!=number_arr.length-1) answer *= 10;
    }
    
    return answer;
}